### PR TITLE
One big chunk of code Pull request

### PR DESCRIPTION
**Pull request by HERMON-1995**
> This code is from To-do-list project to check if the code follow JS best practice.

### CODE REVIEW

File: src/modules/Clear_all.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  let incompleted = localGet().filter((item) => {
+    if (!item.completed) {
+      return item;
+    }
+    return null;
+  });
+
+  incompleted = incompleted.map((item, id) => {
+    item.index = id + 1;
+    return item;
```
Comment (dheerajsachdeva) [inline src/modules/Clear_all.js:13]: Here, you can use the shorthand property like ++

File: src/modules/UI.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+};
+
+window.updateList = (id) => {
+  const updateInput = document.querySelector(`#input-${id}`).value;
+  const updateArray = localGet().map((item) => {
+    if (item.index - 1 === id) {
+      item.description = updateInput;
+
+      if (item.completed === true) {
+        item.completed = false;
```
Comment (dheerajsachdeva) [inline src/modules/UI.js]: You can use shorthand property:  like if(item.completed)

File: src/modules/UI.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        `;
+  });
+};
+
+const addList = (description, completed, index) => {
+  const listAdded = new ListTemplate(description, completed, index);
+  const x = localGet();
+  x.push(listAdded);
+  localStorage.setItem('listStorage', JSON.stringify(x));
+  todoTask.value = '';
```
Comment (dheerajsachdeva) [inline src/modules/UI.js:38]: You can create a function to set items as you are using this at multiple places in code

**Review by dheerajsachdeva — CHANGES_REQUESTED** [review]
Hi @HERMON-1995 
Good job so far 👍🏼
You still need to work on some issues to go to the next project, but you are almost there!

Highlights
✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

Required Changes ♻️
Check the comments under the review.

Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by HERMON-1995** [comment]
Thanks for the review. I have done the required changes.

File: src/modules/UI.js (hunk 3 above)
Comment (HERMON-1995) [reply src/modules/UI.js:38]: Thanks for the review. I have done the required changes.

**Review by dheerajsachdeva — APPROVED** [review]
Hi @HERMON-1995 

Your project is complete 👍 💯 There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

Highlights 👏🏻
✔️ No linter errors.
✔️ You used correct Git Hub-Flow.
✔️ Professional README
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.

Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**